qtdialogs/DlgWalletMigration.py

```diff
@@ -532,117 +484,83 @@ def accept(self):
+      reuseBtn = msgBox.addButton(self.tr(
+         'Reuse Old Passphrase'), QtWidgets.QMessageBox.AcceptRole)
+      newBtn = msgBox.addButton(self.tr(
+         'Create New Passphrase'), QtWidgets.QMessageBox.ActionRole)
+      cancelBtn = msgBox.addButton(self.tr(
+         'Cancel'), QtWidgets.QMessageBox.RejectRole)
+      msgBox.setDefaultButton(reuseBtn)
+      msgBox.exec_()
+      if msgBox.clickedButton() == reuseBtn:
+         return 'reuse'
+      elif msgBox.clickedButton() == newBtn:
+         return 'new'
+      else:
+         return 'cancel'
+
+   # Handle completion of set/verify passphrase pages
+   def nextFromSetPassphrase(self):
+      # Called when setPassphrasePage is complete
+      self._newPassphrase = self.setPassphrasePage.pageFrame.getPassphrase()
+      self.verifyPassphrasePage.pageFrame.edtPasswd3.clear()
+      self.stack.setCurrentWidget(self.verifyPassphrasePage)
+      self.resize(self.stack.currentWidget().sizeHint())
+
+   def nextFromVerifyPassphrase(self):
+      # Called when verifyPassphrasePage is complete
+      verifyPass = self.verifyPassphrasePage.pageFrame.edtPasswd3.text()
+      if verifyPass != self._newPassphrase:
+         QtWidgets.QMessageBox.critical(self, self.tr('Invalid Passphrase'),
+            self.tr(
+               'You entered your confirmation passphrase incorrectly!'), QtWidgets.QMessageBox.Ok)
+         return
+      # Send new passphrase to backend
+      packet = self.getNewPacket()
+      reply = packet.init('walletCreation')
+      reply.passphrase = self._newPassphrase
+      reply.kdfTargetMs = 2000
+      reply.kdfTargetMB = 128
+      packet.success = True
+      self.reply()
+      self.stack.setCurrentWidget(self.walletProgressPage)
+      self.resize(self.stack.currentWidget().sizeHint())
+
+   # Connect page completion signals
+   def showEvent(self, event):
+      super().showEvent(event)
+      # Connect only once
+      try:
+         self._wired
+      except AttributeError:
+         self.setPassphrasePage.pageFrame.editPasswd2.returnPressed.connect(
+            self.nextFromSetPassphrase)
+         self.verifyPassphrasePage.pageFrame.edtPasswd3.returnPressed.connect(
+            self.nextFromVerifyPassphrase)
+         self._wired = True
+
+   def handleUnlockCancelForWatchOnly(self, unlockDlg):
+      reply = QtWidgets.QMessageBox.question(
+         self, self.tr('Create Watching-Only Wallet?'),
+         self.tr('Unlock was cancelled. Do you want to create a watching-only wallet instead?'),
+         QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No
+      )
+      if reply == QtWidgets.QMessageBox.Yes:
+         # Show progress page and trigger backend
+         unlockDlg._suppressUnlockError = True
+         self.stack.setCurrentWidget(self.walletProgressPage)
+         self.resize(self.stack.currentWidget().sizeHint())
+         unlockDlg.reply("")
+         self.proceedWithMigration()
+      else:
+         self.reject()
```

Comment:
same as the other reject, backend needs to know the migration process has to stop